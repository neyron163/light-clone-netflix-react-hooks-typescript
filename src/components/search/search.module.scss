@import "../variable.scss";

.search {
  display: flex;
  position: relative;
  overflow: hidden;

  .input {
    background: none;
    border: 1px solid var(--color-white);
    font-weight: var(--font-weight-bold);
    padding: 8px 35px;
    color: var(--color-white);
  }

  .magnifierButton {
    cursor: pointer;
    position: absolute;
    left: 10px;
    top: 50%;
    z-index: 2;
    transform: translateY(-50%);
  }

  .magnifier {
    width: 15px;
    height: 19px;
  }

  .animate {
    transform: translateX(100%);
    transition: 0.3s transform;
  }

  .visible {
    transform: translateX(0);
  }
}
