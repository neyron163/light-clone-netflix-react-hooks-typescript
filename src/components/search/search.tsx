import React, { useState, useCallback } from "react";
import magnifier from "../../images/magnifier.svg";
import classNames from 'classnames';
import s from "./search.module.scss";

interface PropsType {
  onChange(e: React.ChangeEvent<HTMLInputElement>): void;
  result: string;
}

export const Search = ({ onChange, result }: PropsType) => {
  const [visible, setVisible] = useState(false);

  const onClick = useCallback(() => {
    setVisible(!visible);
  }, [visible]);

  return (
    <div className={s.search}>
      <button className={s.magnifierButton} onClick={onClick}>
        <div className={s.magnifier}>
          <img src={magnifier} alt="magnifier" />
        </div>
      </button>
      <div className={classNames(s.animate, {[s.visible]: visible})}>
        <input
          className={s.input}
          type="text"
          onChange={e => onChange(e)}
          placeholder="Title, people, genres"
        />
      </div>
    </div>
  );
};
