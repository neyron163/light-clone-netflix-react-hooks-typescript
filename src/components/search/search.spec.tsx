import React from "react";
import renderer from "react-test-renderer";

import { Search } from "./search";

let props;

beforeEach(() => {
  props = {
    onChange: jest.fn()
  };
});

describe("<Search />", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Search {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
