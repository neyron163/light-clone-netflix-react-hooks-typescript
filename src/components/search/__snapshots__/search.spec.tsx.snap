// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render correctly 1`] = `
<div
  className="search"
>
  <button
    className="magnifierButton"
    data-testid="button"
    onClick={[Function]}
  >
    <div
      className="magnifier"
    >
      <img
        alt="magnifier"
        src="magnifier.svg"
      />
    </div>
  </button>
  <div
    className="animate"
    data-testid="animate"
  >
    <input
      className="input"
      data-testid="input"
      onChange={[Function]}
      placeholder="Title, people, genres"
      type="text"
    />
  </div>
</div>
`;
